docs/refs/api/rtos/callbacks_sonar_example.md

```diff
@@ -0,0 +1,106 @@
+### Sonar Example
+
+Here is an example that uses everything discussed in the form of a minimal Sonar class.
+
+```c++
```

Comment:
Can this sample be made into a repository so that it can be edited without changes to the docs?

Reply:
Hmm, probably, do examples each get their own repos?

Reply:
Might be a question for @AnotherButler. Would you prefer a link to a repo or a markdown file with code?

Reply:
https://developer.mbed.org/teams/mbed_example/code/callback-sonar-example/